src/app/common/theme-provider.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type ComputedTheme = 'Dark' | 'Light';
+export type UserSelectedTheme = ComputedTheme | 'System';
+
+const ThemeContext = createContext<
+  [ComputedTheme, UserSelectedTheme, Dispatch<SetStateAction<UserSelectedTheme>>]
+>([
+  // These values are not used, but are set to satisfy the context's value type.
+  'Light',
+  'System',
+  () => {},
+]);
+
+const getSystemTheme = () =>
+  window.matchMedia('prefers-color-scheme: dark').matches ? 'Dark' : 'Light';
+
+export const ThemeSwitcherProvider: FC = ({ children }) => {
+  const [userSelectedTheme, setUserSelectedTheme] = useState<UserSelectedTheme>('System');
+  const [theme, setTheme] = useState<ComputedTheme>(getSystemTheme());
+
+  useEffect(() => {
+    switch (userSelectedTheme) {
+      case 'System': {
```

Comment:
Maybe make a shared enum for these in the `constants` file to avoid possible misspellings?

Reply:
I see this is a type from Kyran's comment so will resolve, but agree to lowercase.